src/edu/csus/ecs/pc2/clics/API202306/ProblemService.java

```diff
@@ -0,0 +1,138 @@
+@Produces(MediaType.APPLICATION_JSON)
+@Provider
+@Singleton
+public class ProblemService implements Feature {
+
+    private IInternalContest model;
+
+    @SuppressWarnings("unused")
+    private IInternalController controller;
+
+    private JSONTool jsonTool;
+
+    public ProblemService(IInternalContest inContest, IInternalController inController) {
+        super();
+        this.model = inContest;
+        this.controller = inController;
+        jsonTool = new JSONTool(inContest, inController);
+    }
+
+    /**
+     * This method returns a representation of the current contest problems in JSON format. The returned value is a JSON array with one problems description per array element, compying with 2023-06
+     * 
+     * @param sc User information
+     * @param contestId The contest
+     * @return a {@link Response} object containing the contest problems in JSON form
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getProblems(@Context SecurityContext sc, @PathParam("contestId") String contestId) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == false) {
+            return Response.status(Response.Status.NOT_FOUND).build();        
+        }
+        // get the problems from the contest
+        Problem[] problems = model.getProblems();
+        int ord = 1;
+        ArrayList<CLICSProblem> problist = new ArrayList<CLICSProblem>();
+        
+        // public only gets the problems when the contest starts
+        if (sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) || sc.isUserInRole(WebServer.WEBAPI_ROLE_JUDGE) || model.getContestTime().isContestStarted()) {
+            for (Problem problem: problems) {
+                if (problem.isActive()) {
+                    problist.add(new CLICSProblem(model, problem, ord));
+                    ord++;
+                }
+            }
+        }
+        try {
+            ObjectMapper mapper = JSONUtilities.getObjectMapper();
+            String json = mapper.writeValueAsString(problist);
+            return Response.ok(json, MediaType.APPLICATION_JSON).build();
+        } catch (Exception e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Error creating JSON for problems " + e.getMessage()).build();
+        }
+    }
+
+    /**
+     * Return the reponse to a request for a single problem's information for the specified contest and problem id.
+     * 
+     * @param sc user info
+     * @param contestId the contest
+     * @param problemId the problem id desired
+     * @return response
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{problemId}/")
+    public Response getProblem(@Context SecurityContext sc, @PathParam("contestId") String contestId, @PathParam("problemId") String problemId) {
+        // get the problems from the contest
+        Problem[] problems = model.getProblems();
+        int ord = 1;
+        for (Problem problem: problems) {
```

Comment:
if a guest guesses the problemId would they not get access to this problem?  that seems bad.

Reply:
Oooo, yes.  I forgot to add the permission check like I did for when the entire list is queried.  Fixed in next push.

Reply:
This has been pushed.  Please resolve this comment.